lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -519,131 +644,135 @@ def _rows_for_users(self, context, users):
+                        subsection_index=subsection_index,
+                        subsection_name=subsection_info['subsection_block'].display_name,
+                    )
+                    scorable_blocks_map[scorable_block.location] = [header_name + " (Earned)",
+                                                                    header_name + " (Possible)"]
+        return scorable_blocks_map
 
-        for student, course_grade, error in CourseGradeFactory().iter(enrolled_students, course):
+    def _rows_for_users(self, context, users, header_row, graded_scorable_blocks):
+        """
+        Returns a list of rows for the given users for this report.
+        """
+        success_rows, error_rows = [], []
+        for student, course_grade, error in CourseGradeFactory().iter(users, context.course):
             student_fields = [getattr(student, field_name) for field_name in header_row]
-            task_progress.attempted += 1
 
             if not course_grade:
                 err_msg = text_type(error)
                 # There was an error grading this student.
                 if not err_msg:
-                    err_msg = u'Unknown error'
+                    err_msg = 'Unknown error'
                 error_rows.append(student_fields + [err_msg])
-                task_progress.failed += 1
                 continue
 
-            enrollment_status = _user_enrollment_status(student, course_id)
+            enrollment_status = _user_enrollment_status(student, context.course_id)
 
             earned_possible_values = []
             for block_location in graded_scorable_blocks:
                 try:
                     problem_score = course_grade.problem_scores[block_location]
                 except KeyError:
-                    earned_possible_values.append([u'Not Available', u'Not Available'])
+                    earned_possible_values.append(['Not Available', 'Not Available'])
                 else:
                     if problem_score.first_attempted:
                         earned_possible_values.append([problem_score.earned, problem_score.possible])
                     else:
-                        earned_possible_values.append([u'Not Attempted', problem_score.possible])
-
-            rows.append(student_fields + [enrollment_status, course_grade.percent] + _flatten(earned_possible_values))
+                        earned_possible_values.append(['Not Attempted', problem_score.possible])
 
-            task_progress.succeeded += 1
-            if task_progress.attempted % status_interval == 0:
-                step = u'Calculating Grades'
-                task_progress.update_task_state(extra_meta={'step': step})
-                log_message = u'{0} {1}/{2}'.format(step, task_progress.attempted, task_progress.total)
-                log_task_info(log_message)
+            success_rows.append(
+                student_fields + [enrollment_status, course_grade.percent] + _flatten(earned_possible_values))
```

Comment:
`task_progress.succeeded += 1`

Reply:
There was also a status update after the `status_interval` we are also missing that, can you explain is this intentional and why?